std/range/package.d

```diff
@@ -7130,6 +7130,17 @@ sgi.com/tech/stl/binary_search.html, binary_search).
         }
         return false;
     }
+
+// groupBy
+/**
+Returns a range of subranges of elements that are equivalent according to the
+sorting relation.
+ */
+    auto groupBy()()
```

Comment:
Andrei suggested the name. He originally wanted 'group' but that would cause UFCS issues with std.algorithm.group. Why group instead of chunk? Here's how I think about it:

r = [1a, 1b, 2c, 1d]
r grouped = [[1a, 1b, 1d], [2c]]
r chunked = [[1a, 1b], [2c], [1d]]

i.e. chunking only handle adjacent elements, whereas grouping handles all elements. With sorted range, we can easily guarantee the group behaviour.